pyccl/tracers.py

```diff
@@ -264,6 +265,157 @@ def get_bessel_derivative(self):
 
         return np.array([t.der_bessel for t in self._trc])
 
+    def _MG_add_tracer(self, cosmo, kernel, mg_transfer,
+                       k=None, der_bessel=0, der_angles=0,
+                       bias_transfer_a=None, bias_transfer_k=None):
+        """ function to set mg_transfer in the right format and add MG tracers
+            for different cases including different cases and biases like
+            intrinsic alignements (IA) when present
+        """
+        # case with no astro biases
+        if ((bias_transfer_a is None) and (bias_transfer_k is None)):
+            if (isinstance(k, (list, np.ndarray))):
+                # Scale-dependent case with full k array
+                self.add_tracer(cosmo, kernel,
+                                transfer_ka=mg_transfer, der_bessel=der_bessel,
+                                der_angles=der_angles)
+            elif (isinstance(k, float) or isinstance(k, int)):
+                # Scale-dependent case with  a single specified k value
+                self.add_tracer(cosmo, kernel,
+                                transfer_a=mg_transfer, der_bessel=der_bessel,
+                                der_angles=der_angles)
+            else:
+                # Scale-independent case where k=None
+                self.add_tracer(cosmo, kernel,
+                                transfer_a=mg_transfer, der_bessel=der_bessel,
+                                der_angles=der_angles)
+
+        #  case of an astro bias depending on a and  k
+        elif ((bias_transfer_a is not None) and (bias_transfer_k is not None)):
+            if (isinstance(k, (list, np.ndarray))):
+                # Scale-dependent case with full k array
+                mg_transfer_new = (mg_transfer[0], mg_transfer[1],
+                                   (bias_transfer_a[1] * (bias_transfer_k[1] *
+                                    mg_transfer[2]).T).T)
+                self.add_tracer(cosmo, kernel,
+                                transfer_ka=mg_transfer_new,
+                                der_bessel=der_bessel, der_angles=der_angles)
+            elif (isinstance(k, float) or isinstance(k, int)):
+                # Scale-dependent case with  a single specified k value
+                mg_transfer_new = (mg_transfer[0], mg_transfer[1] *
+                                   bias_transfer_a[1])
+                self.add_tracer(cosmo, kernel, transfer_a=mg_transfer_new,
+                                der_bessel=der_bessel, der_angles=der_angles,
+                                transfer_k=bias_transfer_k)
+            else:
```

Comment:
Can this case be merged with the previous one? I think all the lines are the same.

Reply:
condensed